app/src/main/java/com/example/ubclaunchpad/launchnote/edit/PhotoInfoActivity.kt

```diff
@@ -35,25 +33,47 @@ class PhotoInfoActivity : AppCompatActivity() {
     private lateinit var recyclerView: RecyclerView
     private lateinit var viewAdapter: RecyclerView.Adapter<*>
     private lateinit var viewManager: RecyclerView.LayoutManager
-
-    private var classButtons = arrayOf("CPSC 213", "CPSC 313", "CPSC 406",
-            "CPSC 213", "CPSC 313", "CPSC 406", "CPSC 213",
-            "CPSC 313", "CPSC 406", "CPSC 213", "CPSC 313",
-            "CPSC 406", "CPSC 213", "CPSC 313", "CPSC 406").map { ClassButtonModel(it) }.toTypedArray()
+    private val classButtons : MutableList<ClassButtonModel> = mutableListOf()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_photo_info)
         ButterKnife.bind(this)
 
         savedInstanceState?.let {
-            classButtons = it.getParcelableArray(LIST_OF_CLASSES_KEY) as Array<ClassButtonModel>
+            classButtons.addAll(it.getParcelableArray(LIST_OF_CLASSES_KEY) as Array<ClassButtonModel>)
+        } ?: run {
+            grabButtonsFromDb( {viewAdapter.notifyDataSetChanged()})
         }
 
         picNoteToEdit = intent.extras.getSerializable(PIC_NOTE_ARG) as PicNote
         removeIfNoSave = intent.extras.containsKey(REMOVE_IMAGES_IF_NO_CHANGE) &&
                 intent.extras.getBoolean(REMOVE_IMAGES_IF_NO_CHANGE)
 
+        setTextFields()
+        setFolderRecyclerView()
+        setListeners()
+    }
+
+    private fun grabButtonsFromDb(whenDone: () -> Unit) {
+        LaunchNoteDatabase.getDatabase(this)?.let {
+            it.folderDao().loadAll().map({
```

Comment:
naming one of the "it"s would make this part clearer, like
`it.folderDao().loadAll().map{ folders ->
                folders.map { ClassButtonModel(it.name, it.id, picNoteToEdit.folderId == it.id) }
            )`